### PR TITLE
working axios with API. added some basic styles to countries list

### PR DESCRIPTION
**Pull request by marcinel123**
Hi Adam,

In this PR:
- added api and axios to the project
- also I added some basic styling to countries list so we can see if all contries works fine.

Martin

### CODE REVIEW

File: src/components/singleCountry/SingleCountry.styles.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import styled from "styled-components";
+
+export const StyledDivWithCountryData = styled.div`
```
Comment (AdamKniec) [inline src/components/singleCountry/SingleCountry.styles.ts]: pick a different name here 👍

File: src/components/CountriesList/CountriesList.tsx [hunk 2]
```diff
@@ -1,5 +1,41 @@
...
+		};
+		fetchCountries();
+	}, []);
+
+	return (
+		<>
+			{error && <p>There is an error fetching API.</p>}
+			<StyledDisplaySection>
+				{countries
+					? countries.map(({ name, flags, population, region, capital }) => (
```
Comment (AdamKniec) [inline src/components/CountriesList/CountriesList.tsx]: Google for Early return pattern and use it here 👍

File: src/components/CountriesList/CountriesList.tsx [hunk 3]
```diff
@@ -1,5 +1,41 @@
...
+							<SingleCountry
```
Comment (AdamKniec) [inline src/components/CountriesList/CountriesList.tsx]: I propose CountryCard

File: src/components/CountriesList/CountriesList.tsx [hunk 4]
```diff
@@ -1,5 +1,41 @@
...
+import { SingleCountry } from "../singleCountry/SingleCountry";
 
-export const CountriesList = () => (
-	<StyledDisplaySection>CountriesList</StyledDisplaySection>
-);
+export const CountriesList = () => {
+	const [error, setError] = useState<boolean>();
+	const [countries, setCountries] = useState<CountriesProps[]>();
+
+	useEffect(() => {
```
Comment (AdamKniec) [inline src/components/CountriesList/CountriesList.tsx:9]: Make a note to yourself to refactor the in the future to extract this logic to some different place

File: src/components/singleCountry/SingleCountry.styles.ts (hunk 1 above)
Comment (marcinel123) [reply src/components/singleCountry/SingleCountry.styles.ts]: fixed.

File: src/components/CountriesList/CountriesList.tsx (hunk 3 above)
Comment (marcinel123) [reply src/components/CountriesList/CountriesList.tsx]: fixed.

File: src/components/CountriesList/CountriesList.tsx (hunk 4 above)
Comment (marcinel123) [reply src/components/CountriesList/CountriesList.tsx:9]: I moved fetching api logic to custom hook.

**Review by AdamKniec — APPROVED** [review]
👍

**Pull request merged**